compiler/damlc/tests/daml-test-files/LfStableTimeout.daml

```diff
@@ -0,0 +1,33 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module LfStableTimeout where
+
+import Daml.Script
+
+loop : Int -> ()
+loop x = loop (x + 1)
+
+-- @ERROR range=13:1-13:15; Evaluation timed out
+-- @LEDGER testScriptLoop LfStableTimeout.testScriptLoop.EXPECTED.ledger
+testScriptLoop = script do
+  alice <- allocateParty "p"
+  submit alice do
+     pure $ loop 1
```

Comment:
Note that this behaviour is _slightly_ different in Scripts.
The loop is evaluated by the script runner eagerly, before `submit` tries to do anything.
Dropping the submit would make no difference to whats being evaluated

Whereas in scenarios, i believe the evaluation of the inner Update is lazy
I dont think anything here needs changing, just wanted to point it out :)

Reply:
oh that's a very subtle point. I think it would be nice to extend this test with a `create` in order to show that it was never performed; will do tomorrow

Reply:
interesting, I added a `createCmd` before `pure $ loop 1` expecting the create to never occur but it did, see https://github.com/digital-asset/daml/pull/17375/commits/c55b7b57defdebe9e0162f713502cdbcce1cdc4f

Reply:
Interesting, I guess the final pure is lazy? I wonder what happens if there was a create then loop then create...
I also wonder if this behaviour changes with daml3-script, since Commands was refactored entirely